### DISCUSSION TITLE
Road to v1

### DISCUSSION THREAD

**Discussion by pom421** [pom421/transparencemedia#4] [General]
Hello,

Je donne ici une vue sur le contexte et ce que j'essaie d'atteindre. 

Maël a fait il y a quelques années un site sur l'actionnariat dans les média : http://transparencemedia.fr/.

Par manque de temps, il ne peut plus le maintenir et a fait une annonce pour une éventuelle reprise sur : https://twitter.com/maeool/status/1472170405720907782.

Je suis intéressé pour que le projet ne tombe pas dans l'oubli, surtout avec la présidentielle qui s'annonce ^^.

De mon côté, je travaille en React et donc je me suis senti capable et curieux de voir comment migrer un projet en React assez vieux (React 15, Webpack 2) et le refactor en projet Next. 

Grâce à Next, on va pouvoir s'abstraire de beaucoup de tooling un peu pénible. Dans le code legacy, tout le code webpack/babel était géré dans l'app. 

On pourra également profiter des avancées de Next : server side rendering, voire mieux, static site generation. En effet, il n'y a que quelques centaines d'entité (société, journal, individu, etc..), donc ça serait pertinent de récupérer les données de ces entités (via Wikipedia) au build time, ce qui donnerait un site ultra rapide.

Bref, pour voir le détail de ce qui se passe, on peut se reporter à l'issue #1 et à la PR en cours #2.

Mon but pour la v1, est de rendre le site à peu près fonctionnel et à combler la dette technologique.

Donc il faut que marche : 
- la home page ✅ 
- la page de kiosque qui permet de filtrer par media ✔️ 
- la page de détail d'un journal (en cours)
- le menu
- le layout des pages statiques (contribuer, crédit, etc..)

Côté tooling, j'ai ajouté la configuration ESLint + Prettier + l'auto save dans VS code. 
Le déploiement se fait sur Vercel de façon automatique.

## Les besoins

Si tu te sens concerné par le projet, n'hésite pas à te manifester :).
C'est la première fois que je fais de l'open source donc tout conseil ou aide sera précieuse. 

Voici ce que je vois pour la suite de la roadmap, à affiner : 
- améliorer l'UI, l'UX de l'existant
- refactor en React récent (hook)
- améliorer l'affichage de l'univers (mais déjà fait par le Monde diplo, donc est-ce utile?)
- utiliser les ressources du repo du Monde diplo, comme source de vérité : https://github.com/mdiplo/Medias_francais

**Comment by vict0rsch** [comment]
Hello @pom421 ! Super que tu reprennes ce projet. Le dev web n'est pas mon expertise, j'avais développé en amateur le projet Metada https://metada.org au sein d'une organisation que j'avais créée mais que je n'ai jamais trop poussée DataForNews https://github.com/datafornews/metada

Je serai ravi de t'aider pour l'expérience et éventuellement tester des pre-releases, te donner du feedback. Malheureusement comme je bosse à côté et participe à d'autres projet open-source je serai la personne qui m'énervait moi-même il y a quelques années: je peux te dire des choses mais pas les faire...

**Reply by pom421** [reply]
Cool ! Tes connaissances du métier seront très précieuses. Oui, voyons où ça nous mène, avec le temps et l'envie que chacun peut consacrer.

**Reply by laem** [reply]
Je suis dans le même cas :/ 
Dev web mais déjà bien investi sur un projet pro et sur un projet perso assez important où je suis tout seul. 

Mais dispo pour aider à le diffuser :)

**Comment by vict0rsch** [comment]
Par ailleurs je pense que c'est plutôt v2 mais on pourrait discuter de l'idée de navigation dans le graphe comme Metada le fait.

**Reply by pom421** [reply]
Oui, j'ai quelques idées moi aussi, sur comment passer d'entité en entité.
Ce qui m'intéresse dans le projet à l'heure actuel, c'est la visualisation avec le Treemap pour voir le pourcentage d'actionnaires.
J'y connais rien en visualisation, comment rendre ça plus concret et ça va me changer les idées par rapport à du dev web traditionnel (api & db oriented).

**Reply by pom421** [reply]
Et pour répondre, oui je repousse ça à une éventuelle v2. Dans la v1, il faut corriger les bugs, améliorer le visuel, rendre le menu visible (il est KO sur certaines pages et toujours en blanc), rendre ça responsive...

**Reply by laem** [reply]
Ça fait longtemps que je l'ai fait mais il me semble que quelque part dans le code il y avait une vue graphe... j'essaie de vous retrouver ça.

**Reply by laem** [reply]
Ah ben oui c'est le bouton noir ^^
![Peek 23-12-2021 11-13](https://user-images.githubusercontent.com/1177762/147225597-31cbd8a8-d412-42a4-a774-7420b4db87d5.gif)

**Comment by vict0rsch** [comment]
Et puis une des choses qui m'ont menées à créer Metada (et qui a poussé Possedex aussi) c'est qu'un site "passif" c'est bien, une extension pour navigateur "active" c'est encore mieux. 

Là encore, v2 je pense?

**Reply by pom421** [reply]
Par active, tu veux parler de reconnaitre le site sur lequel on est et afficher des informations pertinentes?

**Reply by vict0rsch** [reply]
(sorry, noël toussa toussa)

Oui, je veux dire une extension qui détecte le site sur lequel tu es et qui, si c'est un média connu dans ses données, envoie une notif' pour te rappeler qui sont les "propriétaires finaux" de ce média.

Après l'extension peut aussi avoir un popup qui n'est autre qu'une page html statique et qui peut donc aussi afficher le graphe du média et autres infos pertinentes!

**Comment by camillem** [comment]
Avez-vous une idée de la licence Open Source selon laquelle vous allez diffuser le code / les données ?

**Reply by pom421** [reply]
Non. Des suggestions?
